OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs

```diff
@@ -118,7 +118,7 @@ public void UpdateMouseover()
 			}
 
 			var frozen = world.ScreenMap.FrozenActorsAt(world.RenderPlayer, worldRenderer.Viewport.ViewToWorldPx(Viewport.LastMousePos))
-				.Where(a => a.TooltipInfo != null && a.IsValid)
+				.Where(a => a.TooltipInfo != null && a.IsValid && (a.Actor != null && !a.Actor.IsInvisibleTo(world.RenderPlayer)))
```

Comment:
The brackets don't change the logic here. ```a && (b && c)``` is equivalent to ```a && b && c```. All three conditions must be true for the whole thing to be true.